### PR TITLE
Interior vehicle data rate limitation unit tests

### PR DESCRIPTION
**Pull request by ZhdanovP**
This PR is **ready** for review.

### Testing Plan
ATF tests

### Summary
- Unit tests were fixed
- One unit test was added for limitation
- Limitation request timer was changed

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc [hunk 1]
```diff
@@ -88,6 +94,22 @@ void InteriorDataCacheImpl::ClearCache() {
 void InteriorDataCacheImpl::ResetRequestCountOnTimer() {
   LOG4CXX_AUTO_TRACE(logger_);
   amount_of_request_in_this_time_frame_ = 0;
+  std::this_thread::sleep_for(
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc]: @ZhdanovP why do you sleep in the body of the timer? The timer will periodically call this function. Periodically means once a time frame. We do not need to sleep

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc [hunk 2]
```diff
@@ -47,8 +47,8 @@ bool RCRPCPlugin::Init(
     policy::PolicyHandlerInterface& policy_handler) {
   const auto& settings = app_manager.get_settings();
   const auto frequency = settings.get_vehicle_data_frequency();
-  interior_data_cache_.reset(new InteriorDataCacheImpl(frequency.second));
-
+  interior_data_cache_.reset(new InteriorDataCacheImpl());
+  interior_data_cache_->StartRequestResetTimer(frequency.second);
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc]: @ZhdanovP why can't we start timer in InteriorDataCacheImpl constructor?

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc [hunk 3]
```diff
@@ -88,6 +94,22 @@ void InteriorDataCacheImpl::ClearCache() {
...
+}
+
+void InteriorDataCacheImpl::StartRequestResetTimer(
+    const uint32_t time_frame_of_allowed_requests) {
+  if (!reset_request_count_timer_.is_running()) {
+    reset_request_count_timer_.Stop();
+  }
+
+  time_frame_of_allowed_requests_ = time_frame_of_allowed_requests;
+  const uint32_t translate_msec_to_sec = 1000;
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc]: @ZhdanovP you can fetch this constants from date time

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc (hunk 1 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc]: @LuxoftAKutsan removed ff6979d4bdcbd3dca9dd6a79d9a02e83c4c40788

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc (hunk 3 above)
Comment (LitvinenkoIra) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc]: @LuxoftAKutsan done ff6979d4bdcbd3dca9dd6a79d9a02e83c4c40788

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc (hunk 2 above)
Comment (ZhdanovP) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_rpc_plugin.cc]: @LuxoftAKutsan it's better to move start in separate method for two reasons: 
1. It's allow to change time frame without killing the interior cache object
2. It's allow to test this method in unit tests

**Pull request merged**